### PR TITLE
Update cypress and cover important analytics events by e2e

### LINKED ISSUE #8698: Analytics - cover important events by e2e
Most important events are covered by E2E tests
    - `suite-ready`
    - `device-connect`

### PR DESCRIPTION
**Pull request by tomasklim**
## Description

- check that `transport-type`, `suite-ready`, `device-connect`, `device-disconnect` events are reported with correct values in separate file
- update user agent strings for unsupported browsers to be more up-to-date
- add checks of other analytics events to existing tests
- add eslint plugin lib to do not say that code like this is incorrect 
`expect(suiteReadyEvent.screenWidth).to.exist.and.not.to.be.empty;`
- last commit creates `cy.task` from shared logic

## Related Issue

closes #8698

## Screenshots:

Screenshot 2023-06-14 at 14 39 48

### CODE REVIEW

**Comment by tomasklim** [comment]
Moving to In progress as I am adding some more events to older e2e

File: packages/suite-web/e2e/tests/analytics/events.test.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        cy.getTestElement('@settings/menu/wallet').click();
+        cy.getTestElement('@settings/wallet/network/btc').click();
+        cy.getTestElement('@settings/wallet/network/eth').click();
+        cy.getTestElement('@settings/wallet/network/tgor').click();
+
+        // custom eth backend
+        cy.getTestElement('@settings/wallet/network/eth/advance').click();
+        cy.getTestElement('@settings/advance/select-type/input').click();
+        cy.getTestElement('@settings/advance/select-type/option/blockbook').click();
+        cy.getTestElement('@settings/advance/url').type('https://eth.marek.pl/');
```
Comment (tomasklim) [inline packages/suite-web/e2e/tests/analytics/events.test.ts:128]: Sounds like a nice domain for ETH node @marekrjpolak

Comment (marekrjpolak) [reply packages/suite-web/e2e/tests/analytics/events.test.ts:128]: ⠀⠀⠀⠀⠀⠀⣴⣿⡆⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⣠⣦⡄⣿⣿⡇⢠⣶⡄⢀⣤⡀
⢀⣠⡄⣿⣿⣧⣿⣿⣧⣿⣿⣧⣼⣿⣿
⣾⣿⡇⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⣿⣿⣧⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡟
⠈⠻⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠟⠁
⠀⠀⠀⣿⣿⣿⣿⣿⣿⣿⣿⣿⡇

File: packages/suite-web/e2e/support/types/index.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { SuiteAnalyticsEvent } from '@trezor/suite-analytics';
```
Comment (mroz22) [inline packages/suite-web/e2e/support/types/index.ts:1]: should go to dev dependencies?

**Review by mroz22 — APPROVED** [review]
crazy stuff, analytics everywhere

File: packages/suite-web/e2e/support/types/index.ts (hunk 2 above)
Comment (tomasklim) [reply packages/suite-web/e2e/support/types/index.ts:1]: Screenshot 2023-06-20 at 13 38 19
added

**Pull request merged**